### ISSUE TITLE
Test higher DRAM space

### ISSUE THREAD

**Issue by dpetrisko** [black-parrot/black-parrot#854]
Currently, our simulation environment tests with 32 bits of DRAM space. We should test with more. This is a little tricky within the confines of VCS, but should be reasonable as a one-off test

**Comment by cknizek** [comment]
Is this still relevant? I'd like to tackle it but only have Verilator access, not VCS. Would that work?

**Comment by dpetrisko** [comment]
Hi, should be fine. The issue with VCS is that you can't instantiate a dense memory of 2**32 bits. Not sure if Verilator has this limitation as well. Regardless, using an sparse/associate/C++-based memory is probably a better approach anyway.

Could you do a quick survey of other open-source projects and see how they handle their backing DRAM? If there's anything generally accepted, we should use that.

So this issue basically has the two prongs:
- enable >32b memory simulations
- test >32b caddr_width configurations

Keep me posted!

**Comment by HUNT-001** [comment]
Hi @dpetrisko, I’d like to work on this if it is still desired.

My plan is to first trace where the current simulation flow effectively limits testing to 32-bit DRAM space, then add a focused one-off test that exercises a higher DRAM address range without disturbing the normal regression flow.

If that sounds good, I’ll start by identifying the current config/test hook and the simplest high-DRAM access pattern to validate.

**Comment by dpetrisko** [comment]
I believe (though you should confirm) that the current dram model does support >32b addresses since it is associative. So this issue would be
- modify e.g. mega parrot config for a large caddr_width_p
- a directed test in BP-tests that accesses this higher range as well as its alias to make sure they are independent

**Comment by HUNT-001** [comment]
Thanks, that helps clarify the scope.

I’ll start by tracing the mega parrot config path for a larger `caddr_width_p`, then I’ll put together a small BP-tests directed test that accesses a higher DRAM address and its alias to verify they remain independent.

I’ll update the issue once I’ve identified the exact config/test hook.

**Comment by HUNT-001** [comment]
I now have a cleaner local result.
-Baseline Verilator flow builds and runs under the default config.
-The default config currently has caddr_width = 32.
-Using a directed bp-tests case with explicit uncached loads/stores to 0x083000000 and 0x183000000, the default config fails as a negative control.
-Using e_bp_unicore_megaparrot_cfg is not a good next step in my current Verilator flow because that config currently dies in build on WIDTHCONCAT-related errors before simulation.

At this point, the most reasonable path seems to be introducing a small default-derived unicore config with only the address-width fields widened enough to make the higher DRAM address part of the normal cacheable DRAM space, rather than relying on megaparrot.

Does that match what you had in mind for this issue, or would you prefer a different existing config as the validation target?

**Comment by dpetrisko** [comment]
Yeah seems fine. As long as you have a test that fails without your change and passes with the change I can integrate it with the existing configs as it makes sense

**Comment by HUNT-001** [comment]
I’ve opened the `bp-tests` PR for the regression.

Locally, I validated the same test in two configurations:

* `e_bp_default_cfg` → fails
* `e_bp_custom_cfg` with a minimal local address-width override → passes

PR: https://github.com/black-parrot-sdk/bp-tests/pull/48

**Comment by dpetrisko** [comment]
I believe this is resolved with @HUNT-001 's PRs. Thank you!
